Vueified Multiple Selection Form Fields (checkboxes, radios, single/multi select dropdowns)

hujambo-dunia opened the pull request:
This PR replaces an older PR ([#14774](https://github.com/galaxyproject/galaxy/pull/14774)) because it retains authorship/ownership from original dev (AlexO).

(Please replace this header with a description of your pull request. Please include *BOTH* what you did and why you made the changes. The "why" may simply be citing a relevant Galaxy issue.)
(If fixing a bug, please add any relevant error or traceback)
(For UI components, it is recommended to include screenshots or screencasts)

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

astrovsky01 committed:
update

most tests

prettier

minor tweak

import fix

nsoranzo committed:
Drop nose

It's deprecated and unmaintained, and is broken under Python 3.10 .

mvdbeek committed:
Use ``galaxy_driver_class`` to create tool shed driver

Use ``autouse`` fixture to configure test driver

mvdbeek committed:
Use local import of sqlalchemy sessions

If we import too early we just get the inital None. Slightly more robust
but still bad.

mvdbeek committed:
Drop scripts/function_tests.py

jdavcs committed:
Do not instantiate MarkdownFormatHelpers

No need for a separate instance or even a class: it's just functions;
also instead of passing references, just import as needed.

jdavcs committed:
Correct file permissions of goldenpath.py (remove exec)

ElectronicBlueberry committed:
add heading classes

ElectronicBlueberry committed:
change heading default sizes

ElectronicBlueberry committed:
add heading component

ElectronicBlueberry committed:
add inline prop

ElectronicBlueberry committed:
add demo page

ElectronicBlueberry committed:
fix bold separator heading

ElectronicBlueberry committed:
change separator heading appearance
remove icon-separator heading

ElectronicBlueberry committed:
fix RuleCollectionBuilder css leak

ElectronicBlueberry committed:
fix and slight changes Heading component

ElectronicBlueberry committed:
fix heading hierarchy Toolbox
fix heading hierarchy admin Home
add Heading icons to admin Home

ElectronicBlueberry committed:
fix history import heading

ElectronicBlueberry committed:
toolBox and historyImport heading adjustments

ElectronicBlueberry committed:
rename sizes to fit bootstrap
undo default heading size changes

ElectronicBlueberry committed:
admin heading adjustments
tool recommendation heading adjustments

ElectronicBlueberry committed:
various heading level adjustments
new hide-element utility class
dataset details css refactor to avoid top-gap

ElectronicBlueberry committed:
various heading level changes

Commit:
Change ToolPanel search to client side, advanced search to backend

jmchilton committed:
Move base tool data infrastructure into tool_util package.

Keep the refgenie stuff in Galaxy's app package because it brings in additional dependencies.

This is a small part of a larger ongoing project to cleanup tool shed dependencies and allow tool shed repository installations with minimal dependencies.

hujambo-dunia committed:
CSS added back from merge conflict

hujambo-dunia committed:
prettier

hujambo-dunia committed:
Refactored set() for readability. Renamed optArray() as formattedOptions() for consistency. Refactor get() for to organizationally match AlexO's comment and improve readability. AlexO original logic left in place, however very like could still use further code consolidation. Test script variable rename to correspond with main script.

assuntad23 commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+    data() {
+        return {
+            // Currently never set?
+            errorMessage: "",
```
```
        // customize properties
        if (input_def.type == "data_column") {
            input_def.error_text = "Missing columns in referenced dataset.";
        }
```

else
```
error_text: "No options available",
```

assuntad23 committed:
using filter is:deleted to display deleted workflows

guerler commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+        defaultValue: {
+            type: [String, Array],
+            required: false,
+            default: null,
```
Default `required` is `false`, so no need to explicitly define both. Defining either `required: true` or `default: ?` is sufficient.

hujambo-dunia replied:
Change made.

guerler commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+    computed: {
+        formattedOptions() {
+            const formattedOptions = [];
+            for (let i = 0, len = this.options.length; i < len; i++) {
```
Recommending using `forEach` here instead of a `for` loop.

hujambo-dunia replied:
Good idea to condense the logic, remove the loops. Replaced with `map()`.

guerler commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+        selectValueMultiple(val) {
+            const selected = [];
+            const selectedValues = this.normalizeSelectedValues(val);
+
```
Similar to the previous PR, recommending to remove the spaces from these functions, particularly if they are small. Also below please use `forEach` or even map reduce.

hujambo-dunia replied:
Good idea to condense the logic, remove the loops. Change made.

guerler commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+                    default: this.options[i][2],
+                };
+            }
+            if (this.multiple == false && this.optional == true) {
```
Should be written as `!this.multiple && this.optional`.

hujambo-dunia replied:
Agreed - change made.

cat-bro committed:
Add breaking tests for container_resolvers index and show

cat-bro committed:
lint

cat-bro committed:
update lib/galaxy/webapps/galaxy/api/container_resolution.py and lib/galaxy/tool_util/deps/views.py

jmchilton committed:
Remove reinstall_repository endpoint and just reinstall with the API.

jmchilton committed:
More simplification of admin_toolshed.

jdavcs committed:
Fix bug in anvio.py: check if dataset before accessing its attribute

cat-bro committed:
format and change name of class and file

cat-bro committed:
format and change name of class and file

mvdbeek committed:
Merge pull request #14807 from cat-bro/test_container_resolvers_api_index_and_show

fix container_resolvers API index and show

nsoranzo committed:
More fixes for ``__DUPLICATE_FILE_TO_COLLECTION__`` tool

- Fix output collection type
- Fix test output collection name
- Number elements identifiers starting from 1
- Add new HDAs to history, fix traceback:

  ```
  ERROR    galaxy.tools:__init__.py:1922 Exception caught while attempting to execute tool with id '__DUPLICATE_FILE_TO_COLLECTION__':
  Traceback (most recent call last):
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/__init__.py", line 1900, in handle_single_execution
      rval = self.execute(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/__init__.py", line 1995, in execute
      return self.tool_action.execute(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/model_operations.py", line 83, in execute
      self._produce_outputs(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/model_operations.py", line 108, in _produce_outputs
      tool.produce_outputs(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/__init__.py", line 3685, in produce_outputs
      output_collections.create_collection(
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/__init__.py", line 1014, in create_collection
      check_elements(elements)
    File "/usr/users/ga002/soranzon/software/nsoranzo_galaxy/lib/galaxy/tools/actions/__init__.py", line 1005, in check_elements
      assert value.history is not None or value.history_id is not None
  AssertionError
  ```

dannon committed:
Merge pull request #14812 from nsoranzo/misc_fixes_20221018

2 bug fixes

dannon committed:
Merge pull request #14776 from mahendrapaipuri/add_request_obj_authmanager

Make HTTP request object available to `AuthManager`

davelopez committed:
Address comments from code review

- Remove div left over.
- Set autocomplete in form inputs to `off` instead of invalid value.

jmchilton committed:
Remove unused tool shed makos.

jmchilton committed:
Remove unused render_readme_section in common.mako templates.

itisAliRH committed:
Add alert to improve UX for workflow load, summary and state

martenson committed:
remove unneeded super call

martenson committed:
drop actual workflow contents from the response

martenson committed:
use check_ownership() in place of explicit perm call

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

assuntad23 committed:
creating simple Vue test for restoring deleted WF

assuntad23 committed:
passing disabled prop for Tags

assuntad23 committed:
adding conditional rendering based on deleted attribute, implementing stubbed out onRestore method in both WFList (just refreshes)  and WFDropdown (actually makes the call to restore). onRestore in Dropdown needs to use defined undelete endpoint

martenson committed:
remove unused import

jmchilton committed:
Rebase into cleanup admin toolshed...

jmchilton committed:
Remove unused makos related tool shed.

jmchilton committed:
Remove more unused tool shed mako code.

assuntad23 committed:
removing confirmation message for restoration

hujambo-dunia committed:
Cleaned up data transform functions. Could use additional QA to account for NULL values.

hujambo-dunia committed:
Cleaned up data transform functions. Could use additional QA to account for NULL values.

assuntad23 commented on client/src/components/Form/FormElement.vue:
```diff
@@ -30,7 +30,16 @@
                 :type="type" />
-            <FormColor v-else-if="type == 'color'" :id="id" v-model="currentValue" />
+            <FormSelect
+                v-else-if="type == 'select'"
```
```suggestion
                v-else-if="type == 'select' || type == 'data_column' || type == 'genomebuild' || type == 'group_tag'"
```

hujambo-dunia replied:
@assuntad23 Commited this via github. Be sure to pull the latest from the server to see the change.

assuntad23 commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+<template>
+    <div>
```
Add a conditional here if the is_workflow is true, it should redirect to a _fieldText type.

``` 
        if (input_def.is_workflow) {
            return this._fieldText(input_def);
        }
``` 
^ from `parameters.js`

for now, just direct it back to FormParameter, we can change this when we vueify _fieldText

```
<FormParameter
            v-if="attributes.is_workflow"
            :id="id"
            ref="params"
            v-model="currentValue"
            :data-label="title"
            :type="type"
            :attributes="attributes" />
```

hujambo-dunia replied:
Dependency on PR #13375 - these PRs need to go out together (ideally, merged together beforehand)

martenson committed:
use proper manager method for security check

martenson committed:
update expected response code for workflow test

mvdbeek committed:
Merge pull request #14714 from kxk302/idle_connection_pool_cleaner

Idle connection pool cleaner

guerler committed:
Update lib/galaxy/managers/sharable.py

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

dannon committed:
Merge pull request #14606 from davelopez/deferred_data_2_adapt_invocation_export

Adapt invocation export UI for plugins

dannon committed:
Merge pull request #14687 from guerler/remove_makos

Remove legacy base panel makos

mvdbeek committed:
Merge pull request #14793 from jmchilton/cleanup_ts_controllers

Cleanup More Tool Shed Admin Controllers and Makos

jdavcs committed:
Use create_time instead of dropped update_time

jdavcs committed:
Raise exception in set_meta if PG_VERSION file not found

guerler committed:
Minor clean up and fix of prefixed welcome route

martenson committed:
move more logic to the workflow service

ahmedhamidawan committed:
Changing route and component names (and locations), sorting client search by tool name

jdavcs committed:
Read 5 lines instead of file

bgruening committed:
Merge pull request #14810 from jmchilton/cleanup_ts_controllers_3

More tool shed dependency and dead code cleanup.

martenson committed:
drop dead  import

guerler committed:
Fix apparent errors for probably outdated new user confirmation dialog

jdavcs committed:
Make dataset parameter required

It's never None for this class, so having the parameter as optional is
semantically incorrect. This, however, does violate the Liskov
substitution principle (subclass method signature more strict than
parent definition), but we can fix this (as well as many other such
cases in datatypes) by slightly modifying the object hierarchy. Should
be submitted in a follow-up PR.

jmchilton committed:
Refactor ToolShedRepositoryCache into galaxy.tool_shed.cache

Things required to perform tool shed installs should be galaxy.tool_shed - this module doesn't require anything in galaxy-app - galaxy-data is sufficient.

guerler committed:
Remove unused legacy manage dependency controller

guerler committed:
Remove legacy manage dependency mako

guerler committed:
Also remove unused legacy permissions mako

guerler committed:
Remove redirect mako

davelopez committed:
Merge pull request #14794 from martenson/workflow-undelete

implement undeleting a workflow

ElectronicBlueberry committed:
set WinBox base z-index

assuntad23 commented on client/src/components/Form/Elements/FormSelect.vue:
```diff
@@ -0,0 +1,215 @@
+            {{ errorMessage }}
+        </b-alert>
+        <b-row align-v="center">
+            <b-col v-if="display == 'radio'">
```
display can also be equal to : radiobutton

ElectronicBlueberry committed:
refactor tool help
add heading increment to tool help

ElectronicBlueberry committed:
add test for tool help

guerler committed:
Merge pull request #14826 from guerler/remove_legacy_dependency_mako

Remove legacy dependency manager mako

mvdbeek committed:
Merge pull request #14821 from jdavcs/release2205_updatetime_fix

[22.05] Use create_time instead of dropped update_time JobStateHistory attribute

guerler committed:
Remove redirect function from tool_runner controller

guerler committed:
Remove mount helper for Workflow Editor

guerler committed:
Remove history legacy mount helper

guerler committed:
Remove Page Editor / Display mount helpers

guerler committed:
Remove legacy masthead mount helpers

guerler committed:
Fix redirect usage in tool actions?

guerler committed:
Remove workflow invocations legacy mount helper

guerler committed:
Remove legacy dataset embedding mount helpers

guerler committed:
Merge pull request #14828 from ElectronicBlueberry/window-manager-z-index

Fix Tool form header overlaps with window manager

ahmedhamidawan committed:
Fixing tool search tests

davelopez committed:
Fix export history RO-Crate without workflows

The workflows_directory may not exists if no workflow was invoked in the history.

Includes unit test + small tests refactor

ElectronicBlueberry committed:
refactor heading levels
unify heading sizes

ElectronicBlueberry committed:
remove headings demo

ElectronicBlueberry committed:
add headings documentation

dannon committed:
Merge pull request #14830 from ElectronicBlueberry/tool-help-headings

Fix tool help headings and sizes

ElectronicBlueberry committed:
refactor markdownHelp

ElectronicBlueberry committed:
remove bottom margin of title heading

guerler committed:
Merge pull request #14838 from ElectronicBlueberry/remove-jquery-markdownHelp

Refactor markdownHelp

guerler committed:
Merge pull request #14831 from guerler/remove_redirect_mako

Remove redirect mako from tool runner controller

bgruening committed:
Merge pull request #14835 from davelopez/fix_history_export_rocrate

Fix export history RO-Crate without workflows

bgruening committed:
Merge pull request #14823 from guerler/route_welcome

Fixes welcome panel route in login component

davelopez committed:
Fix galaxy.yml.sample generation

Auto comment out the whole value hierarchy for complex default values.

bgruening committed:
Merge pull request #14822 from jmchilton/tool_util_data

Move base tool data infrastructure into tool_util package.

guerler committed:
Merge pull request #14832 from guerler/remove_mount_helpers

Remove legacy js mount-helpers for makos

guerler committed:
Merge pull request #14811 from jmchilton/remove_reinstall

Remove legacy reinstall endpoint from admin_toolshed.py.

jdavcs committed:
Fix f-string error

galaxybot committed:
Update Python dependencies

jdavcs committed:
Fix linting

nsoranzo committed:
Merge pull request #14844 from galaxybot/dev_auto_update_dependencies

Update Python dependencies

jdavcs committed:
Merge pull request #14817 from jdavcs/dev_datatypes_misc2

Misc. fixes to datatypes modules

dannon committed:
Update lib/galaxy/config/config_manage.py

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

mvdbeek committed:
Merge pull request #14841 from davelopez/fix_config_sample

Fix sample config

ElectronicBlueberry committed:
fix unit tests

ElectronicBlueberry committed:
adjust separator heading gap

ElectronicBlueberry committed:
remove tool card section

ElectronicBlueberry committed:
remove margin for inline headings

ElectronicBlueberry committed:
remove toolcard heading css override

davelopez committed:
Add test for ignored comments

davelopez committed:
Add test for ignored comments

davelopez committed:
Fix skip XML comments when linting

davelopez committed:
Fix skip XML comments when linting

mvdbeek committed:
Merge pull request #14824 from jmchilton/tool_repo_cache

Move ToolShedRepositoryCache into galaxy.tool_shed.

nsoranzo committed:
Add type annotations to ``config_manage.py``

And some code cleanups.

assuntad23 committed:
connecting to undelete endpoint

assuntad23 committed:
fixing test after connecting to endpoint

martenson committed:
update the search help box with new filtering option

ElectronicBlueberry committed:
refactor FormElement

dannon committed:
Test upgrade to bootstrap-vue 2.23.  This uses vue-compat to (hopefully?) achieve compatibility with vue3.

mvdbeek committed:
Merge pull request #14845 from davelopez/fix_linter_with_comments

Fix tool linting with comments

mvdbeek committed:
Fix invocation filtering

Fixes https://github.com/galaxyproject/galaxy/issues/14851

ElectronicBlueberry committed:
move style to component

nsoranzo committed:
Use dnspython to check specifically for an "MX" DNS record

when validating the domain name of the email address for a new user.

Fix https://github.com/galaxyproject/galaxy/issues/14829 .

dannon committed:
Merge pull request #14853 from mvdbeek/fix_invocation_filtering

Fix invocation filtering by stored workflow

nsoranzo committed:
Rewrite `CondaContext._guess_conda_properties()` to use `conda info`

nsoranzo committed:
Require also a recent pyOpenSSL package when installing conda

Fix https://github.com/galaxyproject/galaxy/issues/14848 .

nsoranzo committed:
Add `--strict-channel-priority` to conda create/install commands

Fix https://github.com/galaxyproject/galaxy/issues/12790 .

This is added only if using conda >=4.7.5 , which includes this bug
fix: https://github.com/conda/conda/pull/8819

dannon committed:
Drop confirmation on delete since it's easily revertible now.

dannon committed:
Remove jest testing of workflow deletion confirmation (since we don't confirm anymore)

dannon committed:
Fix html in helpHtml (missing tags mainly)

dannon committed:
Update lib/galaxy/webapps/galaxy/buildapp.py

dannon committed:
Drop accept alert from workflow deletion selenium test -- there is no confirmation now

martenson committed:
wrap FAI icons in spans, so the tooltip triggers properly

guerler committed:
Merge pull request #14622 from ElectronicBlueberry/headings

Headings Changes and Heading Component

dannon committed:
Merge pull request #14816 from assuntad23/6493/feature/undelete-workflows

6493/feature/undelete workflows

nsoranzo committed:
Merge pull request #14847 from nsoranzo/type_annot_config_manage

Add type annotations to ``config_manage.py``

nsoranzo committed:
Try to fall back to the A record if MX fails

cat-bro committed:
add test for /api/container_resolvers/resolve

cat-bro committed:
update lib/galaxy/tool_util/deps/views.py

nsoranzo committed:
Merge pull request #14856 from nsoranzo/release_19.05_fix_14848

[19.05] Require also a recent pyOpenSSL package when installing conda

nsoranzo committed:
Merge branch 'release_19.05' into release_19.09

ElectronicBlueberry committed:
fix tests

nsoranzo committed:
Merge branch 'release_19.09' into release_20.01

nsoranzo committed:
Merge branch 'release_20.01' into release_20.05

ElectronicBlueberry committed:
deprecate attribute props

ElectronicBlueberry committed:
fix showField

ElectronicBlueberry committed:
fix test

jmchilton committed:
Merge pull request #14850 from dannon/bsvue-v3

Test upgrade to bootstrap-vue 2.23.

nsoranzo committed:
Merge branch 'release_20.05' into release_20.09

ElectronicBlueberry committed:
add required and optional hints

ElectronicBlueberry committed:
add test cases for required and optional form elements

ElectronicBlueberry committed:
make required mark inline in workflow tool forms

mvdbeek committed:
Drop broken output.parent logic

⚠ I don't know what this was used for, but none of the dataset-like
models have a children list (was probably a relationship at one point).
We have no coverage at this point:
https://app.codecov.io/github/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/actions/__init__.py
and if you add type hints to `out_data` this breaks because there's no
children attribute on DatasetInstance.

nsoranzo committed:
Merge branch 'release_20.09' into release_21.01

nsoranzo committed:
Merge branch 'release_21.01' into release_21.05

nsoranzo committed:
Merge branch 'release_21.05' into release_21.09

nsoranzo committed:
Merge branch 'release_21.09' into release_22.01

mvdbeek committed:
Merge pull request #14852 from davelopez/22.05_backport_#14845

[22.05] Backport of #14845

mvdbeek committed:
Merge pull request #14862 from mvdbeek/drop_output_parent

Drop broken output.parent logic

mvdbeek committed:
Merge pull request #14859 from cat-bro/more-container-resolvers-api

[22.05] Cherry pick #14807 and add test for /api/container_resolvers/resolve

nsoranzo committed:
Merge branch 'release_22.01' into release_22.05

nsoranzo committed:
Merge branch 'release_22.05' into dev

dannon committed:
Merge pull request #14857 from martenson/workflow-checkmark

make solo fontawesome elements render attached tooltips properly

ElectronicBlueberry committed:
remove unused styles

ElectronicBlueberry committed:
remove unused css class .icon

ElectronicBlueberry committed:
fix isEmpty for edge cases

ElectronicBlueberry committed:
add section indicator to conditionals

nsoranzo committed:
Merge pull request #14854 from nsoranzo/fix_14829

Use dnspython to check specifically for an "MX" DNS record

hujambo-dunia committed:
Update client/src/components/Form/FormElement.vue

Co-authored-by: Assunta DeSanto <assuntad23@gmail.com>

hujambo-dunia committed:
Update client/src/components/Form/FormElement.vue

Co-authored-by: Assunta DeSanto <assuntad23@gmail.com>

mvdbeek committed:
Fix loading subworkflows from editor interface

This probably broke in
https://github.com/galaxyproject/galaxy/pull/13984. There were 3 issues
here: Subworkflows are referenced by Workflow.id, not StoredWorkflow.id,
so we'd build the wrong url. The other issue is that if we push the new url
into the router nothing happens as we re-use the component. Instead we
need to listen to route changes and reload the editor config.
The third problem is that the main editor component loads the workflow
data only in the created hook.

I think this needs further refactoring to not use the controller
endpoint; everything is available through exisiting APIs and can be
cached (toolbox, workflows, tags), but this fixes subworkflow editing
for now.

ElectronicBlueberry committed:
add password type to isEmpty string check

mvdbeek committed:
Release tool-util package 22.1.3

mvdbeek committed:
Start work on galaxy-tool-util 22.1.4

ahmedhamidawan committed:
ToolSearch: add a front-end Whoosh parser, add ontology search

nsoranzo committed:
Fix response encoding when getting tool citations via API

Fix the first author surname in:

https://usegalaxy.eu/api/tools/lotus2/citations

without regressing on https://github.com/galaxyproject/galaxy/issues/9369 .

When decoding the response content, it is better to default to utf-8
than trust the encoding guessed from the content itself.

nsoranzo committed:
Add missing ``unittest_utils`` package to galaxy-tool-util

Fix:

```
ModuleNotFoundError: No module named 'galaxy.tool_util.unittest_utils'
```

in https://github.com/conda-forge/galaxy-tool-util-feedstock/pull/4 .

mvdbeek committed:
Merge pull request #14876 from nsoranzo/release_22.05_fix_api_tools_citations

[22.05] Fix response encoding when getting tool citations via API

mvdbeek committed:
Merge pull request #14877 from nsoranzo/release_22.01_fix_tool-util_package

[22.01] Add missing ``unittest_utils`` package to galaxy-tool-util

mvdbeek committed:
Release tool-util package 22.1.4

mvdbeek committed:
Start work on galaxy-tool-util 22.1.5

nsoranzo committed:
Merge branch 'release_22.01' into release_22.05

nsoranzo committed:
Merge branch 'release_22.05' into dev

jmchilton committed:
Move model/mapping utils into packaged code for reuse by the tool shed.

martenson committed:
Merge pull request #14866 from ElectronicBlueberry/conditional-sections

Visually show which parameters belong to a conditional

ahmedhamidawan committed:
Add jest test for tool search utilities.js

ahmedhamidawan committed:
ToolSearch: Conform to style changes in headings

ahmedhamidawan committed:
ToolSearch: ToolBox header style changes

ahmedhamidawan committed:
Merge branch 'dev' of https://github.com/galaxyproject/galaxy into Tool_search_client_only

martenson committed:
reintroduce decoding for PUT /dataset/set_edit calls

mvdbeek committed:
Merge pull request #14885 from martenson/perm-encoded-14883

reintroduce decoding for PUT /dataset/set_edit calls

wm75 committed:
Simplify head wrapper - same functionality, no pipe

dannon committed:
Adds href to collection download link so users have the ability to copy it.

dannon committed:
Merge pull request #14870 from mvdbeek/fix_subworkflow_editing

Fix loading subworkflows from editor interface

wm75 committed:
Bump wrapper version and declare updates to it WF-safe

nsoranzo committed:
Fix mistakenly overwriting ``self._conda_version``

with conda-build version.

Introduced in commit 0ed66646e1234a93193edbd4b5577c535ed32f99 .

nsoranzo committed:
Fix mistakenly overwriting ``self._conda_version``

with conda-build version.

Introduced in commit 0ed66646e1234a93193edbd4b5577c535ed32f99 .

martenson committed:
catch two possible specific exceptions before throwing 500

martenson committed:
Merge pull request #14897 from dannon/fix-collection-download-href

[22.05] Fix collection download link

nsoranzo committed:
Install latest miniconda and conda

nsoranzo committed:
Retry conda `install` and `create` commands without strict channel priority

Restore the possibility to solve environments with packages present in
multiple channels but whose requested version is not available in the
highest priority one.
The issue applies only when using conda >=4.7.5.

Fix the following error for the
`test/integration/test_resolvers.py::TestCondaResolutionIntegration::test_dependency_install`
integration test:

```
____________ TestCondaResolutionIntegration.test_dependency_install ____________

self = <integration.test_resolvers.TestCondaResolutionIntegration object at 0x7f7009682950>

    def test_dependency_install(self):
        """
        Test installation of GNUPLOT dependency.
        """
        data = GNUPLOT
        create_response = self._post("dependency_resolvers/dependency", data=data, admin=True)
        self._assert_status_code_is(create_response, 200)
        response = create_response.json()
>       self._assert_dependency_type(response)

test/integration/test_resolvers.py:46:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <integration.test_resolvers.TestCondaResolutionIntegration object at 0x7f7009682950>
response = {'cacheable': False, 'dependency_resolver': {'auto_init': True, 'auto_install': False, 'can_uninstall_dependencies': T...nda,defaults', ...}, 'dependency_type': 'conda', 'environment_path': '/tmp/tmp5f_37kj2/conda/envs/__gnuplot@_uv_', ...}
type = 'conda', exact = True

    def _assert_dependency_type(self, response, type="conda", exact=True):
        if "dependency_type" not in response:
            raise Exception(f"Response [{response}] did not contain key 'dependency_type'")
        dependency_type = response["dependency_type"]
        assert dependency_type == type, f"Dependency type [{dependency_type}] not the expected value [{type}]"
        if "exact" not in response:
            raise Exception(f"Response [{response}] did not contain key 'exact'")
>       assert response["exact"] is exact
E       assert False is True

test/integration/test_resolvers.py:150: AssertionError
```

due to the fact that this test tries to install version 4.6 of gnuplot,
while in the highest priority conda-forge channel the oldest version
available is 5.0.5.
When trying to create the environment, conda fails to resolve it, so Galaxy
falls back to create an environment with an unversioned gnuplot, which fails
the exact version test.

Introduced in commit bb000b2403e51b2ecf2a4dfe9d2ac560b333a437 .

nsoranzo committed:
Retry conda `install` and `create` commands without strict channel priority

Restore the possibility to solve environments with packages present in
multiple channels but whose requested version is not available in the
highest priority one.
The issue applies only when using conda >=4.7.5.

Fix the following error for the
`test/integration/test_resolvers.py::TestCondaResolutionIntegration::test_dependency_install`
integration test:

```
____________ TestCondaResolutionIntegration.test_dependency_install ____________

self = <integration.test_resolvers.TestCondaResolutionIntegration object at 0x7f7009682950>

    def test_dependency_install(self):
        """
        Test installation of GNUPLOT dependency.
        """
        data = GNUPLOT
        create_response = self._post("dependency_resolvers/dependency", data=data, admin=True)
        self._assert_status_code_is(create_response, 200)
        response = create_response.json()
>       self._assert_dependency_type(response)

test/integration/test_resolvers.py:46:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <integration.test_resolvers.TestCondaResolutionIntegration object at 0x7f7009682950>
response = {'cacheable': False, 'dependency_resolver': {'auto_init': True, 'auto_install': False, 'can_uninstall_dependencies': T...nda,defaults', ...}, 'dependency_type': 'conda', 'environment_path': '/tmp/tmp5f_37kj2/conda/envs/__gnuplot@_uv_', ...}
type = 'conda', exact = True

    def _assert_dependency_type(self, response, type="conda", exact=True):
        if "dependency_type" not in response:
            raise Exception(f"Response [{response}] did not contain key 'dependency_type'")
        dependency_type = response["dependency_type"]
        assert dependency_type == type, f"Dependency type [{dependency_type}] not the expected value [{type}]"
        if "exact" not in response:
            raise Exception(f"Response [{response}] did not contain key 'exact'")
>       assert response["exact"] is exact
E       assert False is True

test/integration/test_resolvers.py:150: AssertionError
```

due to the fact that this test tries to install version 4.6 of gnuplot,
while in the highest priority conda-forge channel the oldest version
available is 5.0.5.
When trying to create the environment, conda fails to resolve it, so Galaxy
falls back to create an environment with an unversioned gnuplot, which fails
the exact version test.

Introduced in commit bb000b2403e51b2ecf2a4dfe9d2ac560b333a437 .

nsoranzo committed:
Force recomputation of conda properties after conda `install`

conda version and conda-build availability may change after an
`install` in the root environment.

nsoranzo committed:
Force recomputation of conda properties after conda `install`

conda version and conda-build availability may change after an
`install` in the root environment.

nsoranzo committed:
Add type annotation to ``conda_util.py``

Also:
- Small refactorings

nsoranzo committed:
Install and use libmamba solver

nsoranzo committed:
Merge pull request #14882 from nsoranzo/enable_libmamba_solver

Install latest miniconda and conda. Enable libmamba solver

jdavcs committed:
Merge pull request #14881 from jmchilton/package_testing_utils

Move model/mapping utils into packaged code for reuse by the tool shed.

qiagu committed:
Add ludwig report datatype

natefoo committed:
Update cryptography to 38.0.3

martenson committed:
remove the whole trycatch block

exceptions should be handled by fastapi

nsoranzo committed:
Merge pull request #14898 from martenson/colldownload-14872

[22.05] Do not mask collection download exceptions

nsoranzo committed:
Merge pull request #14904 from natefoo/cryptography-38.0.3

[22.05] Update cryptography to 38.0.3

mvdbeek committed:
Merge pull request #14903 from qiagu/ludwig

Add ludwig report datatype

nsoranzo committed:
Merge pull request #13909 from nsoranzo/drop_nose

Drop nose

nsoranzo committed:
Merge pull request #14900 from nsoranzo/release_19.05_backport_14882

[19.05] Backport fixes from #14882

nsoranzo committed:
Merge branch 'release_19.05' into release_19.09

nsoranzo committed:
Merge branch 'release_19.09' into release_20.01

nsoranzo committed:
Merge branch 'release_20.01' into release_20.05

mvdbeek committed:
Merge pull request #14895 from wm75/headWrapper-simplify

Simplify head wrapper - same functionality, no pipe

nsoranzo committed:
Merge branch 'release_20.05' into release_20.09

ElectronicBlueberry committed:
fix title display

nsoranzo committed:
Merge branch 'release_20.09' into release_21.01

nsoranzo committed:
Merge branch 'release_21.01' into release_21.05

nsoranzo committed:
Merge branch 'release_21.05' into release_21.09

nsoranzo committed:
Merge branch 'release_21.09' into release_22.01

nsoranzo committed:
Merge branch 'release_22.01' into release_22.05

nsoranzo committed:
Merge branch 'release_22.05' into dev

dannon committed:
Merge pull request #14707 from ahmedhamidawan/Tool_search_client_only

Change ToolPanel search to client side and Advanced search to backend

dannon committed:
Merge pull request #14861 from ElectronicBlueberry/mark-optional-form-elements

Visible marker to distinguish required/optional tool params

hujambo-dunia committed:
Added radio-input type variation

hujambo-dunia committed:
Added radio-input type variation

hujambo-dunia committed:
Removal of error message at child-component level in favor of parent-component error message

hujambo-dunia committed:
Removal of error message at child-component level in favor of parent-component error message

hujambo-dunia committed:
Merge remote-tracking branch 'origin/select_vueify_v3' into select_vueify_v3
